app/models/akka/entities.scala

```diff
@@ -7,7 +7,11 @@ case class ConnectionInfo(dsId: String, linkName: String, isRequester: Boolean,
                           linkData: Option[String] = None, version: String = "",
                           formats: List[String] = Nil, compression: Boolean = false,
                           linkAddress: String = "", brokerAddress: String = ""
-                          , resultFormat: String = models.rpc.DSAMessageSerrializationFormat.MSGJSON) {
+                          , resultFormat: String = models.rpc.DSAMessageSerrializationFormat.MSGJSON
+                          , tempKey: String = "", sharedSecret: Array[Byte] = null, salt: String = ""
```

Comment:
1. probably it is not a good practice to initialize default "null" value, however I'm not sure... Just for example:
```
val justForExample1 = None: Option[Array[Byte]]
val justForExample2 = Array[Byte]()
```
2. also, might a better way is to wrap tempKey, sharedSecret, salt to some general type, then pass it as a single param ?

Reply:
I agree with 1) and 2)

Reply:
1. Yes, you right, changed to Array[Byte]().
2. I think it is ok for now.